src/interlocked_hl.c

```diff
@@ -305,3 +305,27 @@ IMPLEMENT_MOCKABLE_FUNCTION(, INTERLOCKED_HL_RESULT, InterlockedHL_SetAndWakeAll
     return result;
 
 }
+
+IMPLEMENT_MOCKABLE_FUNCTION(, INTERLOCKED_HL_RESULT, InterlockedHL_DecrementAndWake, int32_t volatile_atomic*, address)
```

Comment:
>InterlockedHL_DecrementAndWake [](http://example.com/codeflow?start=53&length=30)

Missing unit (and int) tests.
Shall wait for that.

Reply:
the tests will come in another PR. let me know if they need to be included in the same PR.

Reply:
Yes, at least the UT we want in the same PR, please. I see there are no int tests for Interlocked_HL. It is Ok to add a new int test project as a separate PR, but at least the UT should be done together with the implementation. That's our process! :-)

Reply:
unit test changes are implemented